### ISSUE TITLE
Parameter --base does not work on windows

### ISSUE THREAD

**Issue by toadslop** [lycheeverse/lychee#1454]
Assume I have a website in folder /my/folder/public. My website has many relative urls.

If I cd into /my/folder and run `lychee --base ./public ./public`. On Linux, this works -- ./public is joined to the working directory and the relative urls and produced correctly and all of the links work where they should.

 On Windows, this is the output:

```
[./public\index.html]:
✗ [ERR] file:///C:/my-svg.svg | Failed: Cannot find file
✗ [ERR] file:///C:/styles.css | Failed: Cannot find file
...
```

It looks like on Windows Lychee is discarding the working directory instead of prepending it.

I also noticed when I checked the Windows documentation for the file protocal, the output looked a bit different than what Lychee rendered as well.  You can check the examples on [this page](https://learn.microsoft.com/ja-jp/previous-versions/windows/internet-explorer/ie-developer/platform-apis/jj710207(v=vs.85)) to see what I'm talking about.

**Comment by mre** [comment]
> It looks like on Windows Lychee is discarding the working directory instead of prepending it.

Hum, no idea why. I don't have a Windows machine to test this.

Have you tried absolute paths as described on the page you linked?

```sh
lychee --base file:///C:/my/folder/public ./public
```

Apart from that, I think we need a better path handling crate, because we manually concatenate some of those paths and there are many edge-cases like yours.

**Comment by mre** [comment]
Related windows bug path: https://github.com/lycheeverse/lychee/issues/972

**Comment by toadslop** [comment]
Hey, thanks for getting back to me. Yeah, I did try with an absolute path and I experienced the same behavior. I attempted with both windows style \ paths and linux style / paths:

lychee --base file:///C:/my/folder/public ./public
lychee --base file:///C:\my\folder\public ./public

My results were still:

```
[./public\index.html]:
✗ [ERR] file:///C:/my-svg.svg | Failed: Cannot find file
✗ [ERR] file:///C:/styles.css | Failed: Cannot find file
...
```

**Comment by toadslop** [comment]
Since I have a Windows device I wouldn't mind looking into the behavior myself if you could help point me to the file in the repo where the path handling is defined.

**Comment by mre** [comment]
Sure, that would be nice. I think the bug is somewhere here: https://github.com/lycheeverse/lychee/blob/4340fcbcbb3f4b0325fd8dd7394f2a02c64b8cc7/lychee-lib/src/utils/path.rs#L43-L70

**Comment by toadslop** [comment]
Have something to report. Will put in a PR once I finalize, but the bug is caused by the behavior of `Path::join`. From the Rust documentation:

```
Creates an owned [PathBuf](vscode-file://vscode-app/c:/Users/bnhei/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) with path adjoined to self.

If path is absolute, it replaces the current path. // <-- THIS

See [PathBuf::push](vscode-file://vscode-app/c:/Users/bnhei/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) for more details on what it means to adjoin a path.
```

A relative URL in an HTML document starts with '/', which is how an absolute URL in a file system path starts. So when you provide a relative url like "/style.css" in your HTML file, Path::join sees this as an absolute filepath and discards the parent directory.

**Comment by toadslop** [comment]
Seems like a fix for this would be to prepend relative URLs with a period before trying to join them to a file system path.

**Comment by toadslop** [comment]
Identified another bug while investigating this as well. No time to write about it now as I have to go to the office now, but will follow up later with details.

**Comment by mre** [comment]
> If path is absolute, it replaces the current path. // <-- THIS

Right! I fell into this trap a few times by now. It's kind of a sharp edge of the standard library.
That's probably the issue. Thanks for investigating. 

Your fix makes sense to me. Thankful for a pull request if you find the time. 😃

**Comment by toadslop** [comment]
Actually, I found a ton of windows bugs when I ran the test suite. Might take me a while to work through them to the point where I can write a proper test for the issue that I found.

**Comment by mre** [comment]
Ouch. Sorry to hear that and thankful for any support.

**Comment by mre** [comment]
If it makes the process easier, you could create separate issues for each one. But only if you think it's useful.

**Comment by toadslop** [comment]
Yeah, probably will. Most of the issues are related to windows filepath handling. For example, the test suites are failing because the tests are assuming that tempdir will return a unix style path, but on windows the path starts with c:/, which gets picked up by request as an invalid scheme.

**Comment by mre** [comment]
@toadslop, we made a few path-related changes in the latest lychee versions. If possible, can you double-check what the current status is on Windows? A list of open issues would be nice. 😊

**Comment by mre** [comment]
I'll go ahead and close this. We can always open specific issues for more issue cases.
@toadslop, thanks for the pull request to fix Windows filepaths.
